mr/src/main/java/org/elasticsearch/hadoop/util/IOUtils.java

```diff
@@ -209,4 +213,33 @@ private static BytesArray unwrapStreamBuffer(InputStream in) {
         }
         return null;
     }
+
+    /**
+     * Convert either a file or jar url into a local canonical file, or null if the file is a different scheme.
+     */
+    public static String toCanonicalFilePath(URL fileURL) throws URISyntaxException, IOException {
```

Comment:
should we have proper java for public methods ? (e.g. \@params , \@returns \@throws )

Reply:
This should be addressed in eb6652a but github isn't marking it as outdated